lib/plugins/package/lib/packageService.js

```diff
@@ -134,24 +134,29 @@ module.exports = {
     );
   },
 
+  absoluteArtifactPath(artifact) {
+    if (path.isAbsolute(artifact)) {
+        return artifact;
+    } else {
+        return path.join(this.serverless.config.servicePath, artifact);
+    }
+  },
+
   packageFunction(functionName) {
     const functionObject = this.serverless.service.getFunction(functionName);
     const funcPackageConfig = functionObject.package || {};
 
     // use the artifact in function config if provided
     if (funcPackageConfig.artifact) {
-      const filePath = path.join(this.serverless.config.servicePath, funcPackageConfig.artifact);
+      const filePath = this.absoluteArtifactPath(funcPackageConfig.artifact);
```

Comment:
Let' simply just use `path.resolve` instead of `path.join` (that alone will ensure that absolute paths are respected)

Reply:
ah, yep this looks good to me. I don't work with node.js often and was initially surprised that path.join didn't work that way by default - more like python's path.join which would have discarded the first absolute path automatically.